sparkmagic/setup.py

```diff
@@ -85,7 +85,7 @@ def version(path):
           'pandas>=0.17.1',
           'numpy',
           'requests',
-          'ipykernel>=4.2.2,<5',
+          'ipykernel>=5.1.0,<6.0',
           'ipywidgets>5.0.0,<8.0',
           'notebook>=4.2,<6.0',
```

Comment:
if you're upgrading `ipykernel` from `4.2` to `5.x`, 
wouldn't you also need to upgrade `notebook` from `4.2` to `5.x` ?

Reply:
When bumping notebook version in test installation it leads to dependency hell of incompatible versions in other packages. Seems the JupyterHub docker image will also need to follow upgrades as well.

Reply:
The kernel version doesn't have to match that of the notebook, as they're designed to be separate systems.